File: python/test/t_Sample_large.py
#! /usr/bin/env python

from __future__ import print_function
from openturns import *

TESTPREAMBLE()

try:
    # We create an empty Sample
    sample = Sample(0, 2)
    sample.setName("EmptySample")
    print("sample=", repr(sample))

    try:
        # We access the element of the sample
        p = sample[0]
        print("p=", repr(p))

        # We should NEVER go here
        # no TestFailed available
        raise
    # except OutOfBoundException, ex :
    except:
        pass

    # We create an small Sample
    sample = Sample(1, 2)
    sample.setName("SmallSample")
    print("sample=", repr(sample))

    # We access the element of the sample
    p = Point(sample[0])
    print("p=", repr(p))
    try:
        # We try to access past the last element of the point
        p0, p1, p2 = p[0], p[1], p[2]
        print("p[0]=", p0, " p[1]=", p1, " p[2]=", p2)
        # We should NEVER go here
        raise
    except:
        pass

    # We create a big Sample
    sample = Sample(1000000, 2)
    sample.setName("BigSample")

    # We populate the sample
    for i in range(sample.getSize()):
        sample[i, 0] = i
        sample[i, 1] = i

    print("sample first point=", repr(sample[0]))
    print("sample last  point=", repr(sample[sample.getSize() - 1]))

# except TestFailed, ex :
except:
    import sys
    print("t_Sample_large.py", sys.exc_info()[0], sys.exc_info()[1])

File: lib/test/t_Sample_large.cxx
//                                               -*- C++ -*-
/**
 *  @brief The test file of class Sample of big size
 *
 *  Copyright 2005-2023 Airbus-EDF-IMACS-ONERA-Phimeca
 *
 *  This library is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU Lesser General Public License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  (at your option) any later version.
 *
 *  This library is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU Lesser General Public License for more details.
 *
 *  You should have received a copy of the GNU Lesser General Public License
 *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
 *
 */
#include "openturns/OT.hxx"
#include "openturns/OTtestcode.hxx"

using namespace OT;
using namespace OT::Test;

int main(int, char *[])
{
  TESTPREAMBLE;
  OStream fullprint(std::cout);

  try
  {

#ifdef DEBUG_MEMORY
    std::cerr << "Memory usage std=" << GetMemoryUsage() << " full=" << GetFullMemoryUsage() << std::endl;
#endif

    {
      // We create an empty Sample
      Sample sample(0, 2);
      sample.setName("EmptySample");
      fullprint << "sample=" << sample << std::endl;

      try
      {
        // We access the element of the sample
        Point p(sample.at(0));
        fullprint << "p=" << p << std::endl;

        // We should NEVER go here
        throw TestFailed("Exception NOT thrown");
      }
      catch (OutOfBoundException &)
      {
        // Nothing to do
      }
    }

#ifdef DEBUG_MEMORY
    std::cerr << "Memory usage std=" << GetMemoryUsage() << " full=" << GetFullMemoryUsage() << std::endl;
#endif

    {
      // We create an small Sample
      Sample sample(1, 2);
      sample.setName("SmallSample");
      fullprint << "sample=" << sample << std::endl;

      // We access the element of the sample
      Point p(sample[0]);
      fullprint << "p=" << p << std::endl;
      try
      {
        // We try to access past the last element of the point
        Point err( sample.at(2) );

        // We should NEVER go here
        throw TestFailed("Exception NOT thrown");
      }
      catch (OutOfBoundException &)
      {
        // Nothing to do
      }
    }

#ifdef DEBUG_MEMORY
    std::cerr << "Memory usage std=" << GetMemoryUsage() << " full=" << GetFullMemoryUsage() << std::endl;
#endif

    {
      // We create a big Sample
      Sample sample(1000000, 2);
      sample.setName("BigSample");

#ifdef DEBUG_MEMORY
      std::cerr << "Memory usage std=" << GetMemoryUsage() << " full=" << GetFullMemoryUsage() << std::endl;
      printMemoryUsage();
#endif

      // We populate the sample
      UnsignedInteger size = sample.getSize();
      for(UnsignedInteger i = 0; i < size; i++)
      {
        sample(i, 0) = i;
        sample(i, 1) = i;
      }

      Point mean = sample.computeMean();

      fullprint << "sample first point=" << sample[0]      << std::endl;
      fullprint << "sample last  point=" << sample[size - 1] << std::endl;
      fullprint << "sample mean  value=" << mean           << std::endl;

#ifdef DEBUG_MEMORY
      std::cerr << "Memory usage std=" << GetMemoryUsage() << " full=" << GetFullMemoryUsage() << std::endl;
#endif
    }

  }
  catch (TestFailed & ex)
  {
    std::cerr << ex << std::endl;
    return ExitCode::Error;
  }


  return ExitCode::Success;
}

